Source/Dafny/DafnyMain.cs

```diff
@@ -41,13 +42,12 @@ public class DafnyFile {
         SourceFileName = filePath;
       } else if (extension == ".dll") {
         isPrecompiled = true;
-        Assembly.ReflectionOnlyLoad("DafnyRuntime");
-        var asm = Assembly.ReflectionOnlyLoadFrom(filePath);
+        var asm = AssemblyDefinition.ReadAssembly(filePath);
         string sourceText = null;
-        foreach (var adata in asm.GetCustomAttributesData()) {
+        foreach (var adata in asm.CustomAttributes) {
           if (adata.Constructor.DeclaringType.Name == "DafnySourceAttribute") {
             foreach (var args in adata.ConstructorArguments) {
-              if (args.ArgumentType == System.Type.GetType("System.String")) {
+              if (args.Type.Name == "String") {
```

Comment:
Would this catch any class/interface named `String`? (Note that C#'s keyword is `string` whereas `String` is not reserved.)

Reply:
Good callout, I'll double-check